### ISSUE TITLE
Changing resolution on Linux fails silently

### ISSUE THREAD

**Issue by wingated** [BYU-PCCL/HolodeckPythonBinding#7]
Changing: 

UrbanCity UAVAgent uav0 512 512 PrimaryPlayerCamera OrientationSensor LocationSensor VelocitySensor IMUSensor

to

UrbanCity UAVAgent uav0 480 640 PrimaryPlayerCamera OrientationSensor LocationSensor VelocitySensor IMUSensor

results in a silent hang - looks like the holodeck subprocess has died.

**Comment by joshgreaves** [comment]
I recreated the bug with logging and found these lines:
[2018.01.10-01.08.25:449][  0]LogTemp: Warning: Getting buffer of size 1048576
[2018.01.10-01.08.25:449][  0]LogHolodeck: Subscribing sensor PrimaryPlayerCamera for uav0

It seems 1048576 should be height * width * sizeof(FColor).
FColor is 32 bits so that size is wrong: 480 * 640 * 32 = 9830400 so the allocated buffer is far too small.

Also, we need to have functionality to change the buffer size in Python. @mitchneutron and I discussed changing enums to individual sensor classes with member functions, instead of dictionaries from enum to value.

Handing over to @mitchneutron since the PrimaryPlayerCamera buffer is something he worked on before.
